### PR TITLE
Add collection load cache in proxy

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

### CODE REVIEW

File: internal/proxy/meta_cache.go [hunk 1]
```diff
@@ -186,6 +189,71 @@ func (m *MetaCache) GetCollectionSchema(ctx context.Context, collectionName stri
 	return collInfo.schema, nil
 }
 
+func (m *MetaCache) IsCollectionLoaded(ctx context.Context, msgID UniqueID, collectionName string) error {
+	m.mu.RLock()
+	collInfo, ok := m.collInfo[collectionName]
+	if !ok {
+		m.mu.RUnlock()
+		return fmt.Errorf("cannot find collection info in MetaCache, collectionName = %s", collectionName)
```
Comment (DragonDriver) [inline internal/proxy/meta_cache.go:197]: What if collection not in cache but in meta?

File: internal/proxy/meta_cache.go [hunk 2]
```diff
@@ -186,6 +189,71 @@ func (m *MetaCache) GetCollectionSchema(ctx context.Context, collectionName stri
...
+			},
+		})
+		if err != nil {
+			return err
+		}
+		if showResp.Status.ErrorCode != commonpb.ErrorCode_Success {
+			return errors.New(showResp.Status.Reason)
+		}
+		log.Debug("QueryCoord show collections",
+			zap.Any("collID", collInfo.collID),
```
Comment (DragonDriver) [inline internal/proxy/meta_cache.go:217]: Please use `zap.Int64`, try to avoid using `zap.Any`.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/proxy/meta_cache.go (hunk 1 above)
Comment (bigsheeper) [reply internal/proxy/meta_cache.go]: fixed

File: internal/proxy/meta_cache.go (hunk 2 above)
Comment (bigsheeper) [reply internal/proxy/meta_cache.go]: fixed

**Comment by xiaofan-luan** [comment]
what is load cache? we need a issue about it!

**Comment by bigsheeper** [comment]
> what is load cache? we need a issue about it!

Sorry, my bad. load cache here means load collection state cache in Proxy. As the DQL message stream has been removed, I would make a new pr to merge to master, and close this one.

**Pull request closed without merging**